Fix binary_cross_entropy_with_logits

seayoung1112 opened the pull request:
Summary: binary_cross_entropy_with_logits is broken when redcue is not None

Differential Revision: D10372032

ssnl reviewed (commented):
1. Could you instead use the pattern used in other places, and replace `legacy_get_string` with `legacy_get_enum`?
2. Need some tests


cc @ailzhang this may be why you get different numbers.. It is sooooo bad that our argparser accepted string reduction!

seayoung1112 commented:
@SsnL can we do it in another diff? I'm doing this as a fastest way to unblock our training pipeline from this bug

soumith commented:
@seayoung1112 we should add quick tests in test_nn.py. After that this can be shipped, and you can follow-up on (1) in a later diff.

SsnL commented:
(1) is just a 1-line change and is shorter than your current fix though...

seayoung1112 committed:
Fix binary_cross_entropy_with_logits

Summary: binary_cross_entropy_with_logits is broken when redcue is not None

Differential Revision: D10372032

fbshipit-source-id: 30a9b21623fa2e67f554aa5b587aa285322d2456

ezyang commented:
test is broken. Give it a different name.
